components/scream/src/share/tests/column_ops.cpp

```diff
@@ -131,7 +133,98 @@ TEST_CASE("column_ops_ps_1") {
     }
   }
 
-  SECTION ("mid_to_int_fix_top") {
+  SECTION ("mid_to_int_linear") {
+
+    // Fill v_mid with even numbers and set dz=1, so that v_int should contain odd ones.
+    for (int k=0; k<num_levs; ++k) {
+      v_mid_h(0,k)[0] = 2*(k+1);
+      dz_mid_h(0,k)[0] = 1;
+    }
+    Kokkos::deep_copy(v_mid,v_mid_h);
+    Kokkos::deep_copy(dz_mid,dz_mid_h);
+
+    // Use boundary conditions s.t. v_int becomes monotone increasing odd numbers
+    const Real bc_top = 1;
+    const Real bc_bot = 2*num_levs+1;
+
+    // Run column kernel
+    Kokkos::parallel_for(policy,
+                         KOKKOS_LAMBDA(const member_type& team){
+      const int icol = team.league_rank();
+      auto v_i = ekat::subview(v_int,icol);
+      auto v_m = ekat::subview(v_mid,icol);
+      auto dz  = ekat::subview(dz_mid,icol);
+
+      column_ops::compute_interface_values_linear(team,num_levs,v_m,dz,bc_top,bc_bot,v_i);
+    });
+    Kokkos::fence();
+
+    // Check answer
+    Kokkos::deep_copy(v_int_h,v_int);
+    for (int k=0; k<num_levs+1; ++k) {
+      REQUIRE (v_int_h(0,k)[0] == 2*k+1 );
+    }
+
+    // Re-do with lambda as provider
+    Kokkos::deep_copy(v_int,pack_type(0));
+    Kokkos::parallel_for(policy,
+                         KOKKOS_LAMBDA(const member_type& team){
+      const int icol = team.league_rank();
+
+      auto v_m = [&](const int k)->pack_type {
+        return v_mid(icol,k);
+      };
+      auto dz = [&](const int k)->pack_type {
+        return dz_mid(icol,k);
+      };
+      auto v_i = ekat::subview(v_int,icol);
+
+      column_ops::compute_interface_values_linear(team,num_levs,v_m,dz,bc_top,bc_bot,v_i);
+    });
+    Kokkos::fence();
+
+    // Check answer
+    Kokkos::deep_copy(v_int_h,v_int);
+    for (int k=0; k<num_levs+1; ++k) {
+      REQUIRE (v_int_h(0,k)[0] == 2*k+1 );
+    }
+  }
+
+  SECTION ("int_to_mid_to_int_linear") {
+
+    // Fill v_mid with even numbers and set dz=1, so that v_int should contain odd ones.
+    for (int k=0; k<num_levs+1; ++k) {
+      v_int_h(0,k)[0] = 2*k+1;
+      if (k < num_levs) dz_mid_h(0,k)[0] = 1;
+    }
+    Kokkos::deep_copy(v_int,v_int_h);
+    Kokkos::deep_copy(dz_mid,dz_mid_h);
+
+    // Use boundary conditions s.t. v_int becomes monotone increasing odd numbers
+    const Real bc_top = v_int_h(0,0)[0];
+    const Real bc_bot = v_int_h(0,num_levs)[0];
+
+    // Run column kernel
+    Kokkos::parallel_for(policy,
+                         KOKKOS_LAMBDA(const member_type& team){
+      const int icol = team.league_rank();
+      auto v_i = ekat::subview(v_int,icol);
+      auto v_m = ekat::subview(v_mid,icol);
+      auto dz  = ekat::subview(dz_mid,icol);
+
+      column_ops::compute_midpoint_values(team,num_levs,v_i,v_m);
+      column_ops::compute_interface_values_linear(team,num_levs,v_m,dz,bc_top,bc_bot,v_i);
```

Comment:
I think you need a barrier between the two colum_ops calls, no?

Reply:
Yeah, you're right. Getting lucky with a pass again..

Reply:
I just added them